### PR TITLE
AP_Scripting: logging add units and multipliers

### PR DESCRIPTION
**Pull request by IamPete1**
This adds the ability to log units and multipliers from scripting. 

Works as expected except I'm losing the first two characters of the multipliers, they seem to be OK before they get sent to AP_Logger, any ideas?

### CODE REVIEW

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 1]
```diff
@@ -132,9 +164,10 @@ static int AP_Logger_Write(lua_State *L) {
     const uint64_t now = AP_HAL::micros64();
     luaL_addlstring(&buffer, (char *)&now, sizeof(uint64_t));
 
-    for (uint8_t i=4; i<=args; i++) {
+    for (uint8_t i=feild_start; i<=args; i++) {
```
Comment (tridge) [inline libraries/AP_Scripting/lua_bindings.cpp:167]: welsh?

**Comment by IamPete1** [comment]
There is a bug in the existing code has a buffer overflow. Only noticed now due to it spilling into the multipliers string. The code as it is here works in SITL but not on real hardware (Or a least is not overflowing to where the same place). This is a example of the send texts on Cube Orange. The first is what we expect, the second and third are after doing some stuff with the lua buffer.

```
0: SCR2, Qfff, TimeUS,roll,pitch,yaw, sddd, F---
1: SCR2, Qfff, TimeUS,roll,pitch,yaw, sddd, H0
2: SCR2, Qfff, TimeUS,roll,pitch,yaw, l?@, H0
```

File: libraries/AP_Scripting/lua_bindings.cpp (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_Scripting/lua_bindings.cpp]: fixed

**Comment by IamPete1** [comment]
Thanks to the help of @bugobliterator I have fixed the bug. I was not overflow as I thought, but the units and multipliers going out of  scope. Re-tested in SITL and on real hardware.

Glad to finally got to the bottom of this, only took 3 months....

**Review by tridge — APPROVED** [review]

**Pull request merged**